ArchiSteamFarm.Tests/Bot.cs

```diff
@@ -0,0 +1,386 @@
+				{ (42, Steam.Asset.SteamCommunityContextID, 4), 1 },
+				{ (42, Steam.Asset.SteamCommunityContextID, 7), 1 }
+			};
+
+			AssertResultMatchesExpectation(expectedResult, itemsToSend);
+		}
+
+		[TestMethod]
+		public void OtherAppIDFullSets() {
+			HashSet<Steam.Asset> items = new HashSet<Steam.Asset> {
+				CreateCard(1, realAppID: 42),
+				CreateCard(1, realAppID: 43),
+			};
+
+			HashSet<Steam.Asset> itemsToSend = GetItemsForFullBadge(
+				items, new Dictionary<uint, byte> {
+					{ 42, 1 },
+					{ 43, 1 }
+				}
+			);
+
+			Dictionary<(uint RealAppID, ulong ContextID, ulong ClassID), uint> expectedResult = new Dictionary<(uint RealAppID, ulong ContextID, ulong ClassID), uint> {
+				{ (42, Steam.Asset.SteamCommunityContextID, 1), 1 },
+				{ (43, Steam.Asset.SteamCommunityContextID, 1), 1 }
+			};
+
+			AssertResultMatchesExpectation(expectedResult, itemsToSend);
+		}
+
+		[TestMethod]
+		public void OtherAppIDNoSets() {
+			HashSet<Steam.Asset> items = new HashSet<Steam.Asset> {
+				CreateCard(1, realAppID: 42),
+				CreateCard(1, realAppID: 43),
+			};
+
+			HashSet<Steam.Asset> itemsToSend = GetItemsForFullBadge(
+				items, new Dictionary<uint, byte> {
+					{ 42, 2 },
+					{ 43, 2 }
+				}
+			);
+
+			Dictionary<(uint RealAppID, ulong ContextID, ulong ClassID), uint> expectedResult = new Dictionary<(uint RealAppID, ulong ContextID, ulong ClassID), uint>(0);
+
+			AssertResultMatchesExpectation(expectedResult, itemsToSend);
+		}
+
+		[TestMethod]
+		public void OtherAppIDOneSet() {
+			HashSet<Steam.Asset> items = new HashSet<Steam.Asset> {
+				CreateCard(1, realAppID: 42),
+				CreateCard(2, realAppID: 42),
+				CreateCard(1, realAppID: 43),
+				CreateCard(2, realAppID: 43),
+				CreateCard(3, realAppID: 43),
+				CreateCard(1, realAppID: 44),
+				CreateCard(2, realAppID: 44),
+				CreateCard(3, realAppID: 44),
+				CreateCard(4, realAppID: 44)
+			};
+
+			HashSet<Steam.Asset> itemsToSend = GetItemsForFullBadge(
+				items, new Dictionary<uint, byte> {
+					{ 42, 3 },
+					{ 43, 3 },
+					{ 44, 3 }
+				}
+			);
+
+			Dictionary<(uint RealAppID, ulong ContextID, ulong ClassID), uint> expectedResult = new Dictionary<(uint RealAppID, ulong ContextID, ulong ClassID), uint> {
+				{ (43, Steam.Asset.SteamCommunityContextID, 1), 1 },
+				{ (43, Steam.Asset.SteamCommunityContextID, 2), 1 },
+				{ (43, Steam.Asset.SteamCommunityContextID, 3), 1 }
+			};
+
+			AssertResultMatchesExpectation(expectedResult, itemsToSend);
+		}
+
+		[TestMethod]
+		public void TooHighAmount() {
+			HashSet<Steam.Asset> items = new HashSet<Steam.Asset> {
+				CreateCard(1, 2),
+				CreateCard(2)
+			};
+
+			HashSet<Steam.Asset> itemsToSend = GetItemsForFullBadge(items, cardsPerSet: 2);
+
+			Dictionary<(uint RealAppID, ulong ContextID, ulong ClassID), uint> expectedResult = new Dictionary<(uint RealAppID, ulong ContextID, ulong ClassID), uint> {
+				{ (42, Steam.Asset.SteamCommunityContextID, 1), 1 },
```

Comment:
If you test against fixed constant, then make it a fixed constant instead of relying on constructor to do that for you - everywhere. If you need to specify `42` in all expected results, then it's only counter-intuitive that the constructor for `CreateCard()` generates one with `42` by default, you should probably remove `42` as default value for appID in `CreateCard` and instead declare it explicitly, it no longer makes sense and brings a problem where change of default value will cause tests to fail.

If you ever compare a constant to the same constant in other place, pass constant and compare to constant.